eos/b-decays/lifetime_TEST.cc

```diff
@@ -0,0 +1,257 @@
+ * Copyright (c) 2024 Stefan Meiser
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+#include <eos/observable.hh>
+#include <eos/b-decays/lifetime.hh>
+#include <eos/maths/complex.hh>
+#include <eos/utils/wilson-polynomial.hh>
+
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <fstream>
+#include <limits>
+#include <string>
+#include <vector>
+
+
+using namespace test;
+using namespace eos;
+
+class LifetimeTest :
+    public TestCase
+{
+    public:
+        LifetimeTest() :
+            TestCase("lifetime_test")
+        {
+        }
+
+        virtual void run() const
+        {
+            // SM tests
+            {
+                Parameters p = Parameters::Defaults();
+                p["WET::G_Fermi"]        = 1.1664e-05;
+                p["CKM::abs(V_ud)"]      = 1.000;
+                p["CKM::abs(V_us)"]      = 0.2254;
+                p["CKM::abs(V_cb)"]      = 4.2e-2;
+                p["mass::B_u"]           = 5.279;
+                p["mass::B_d"]           = 5.279;
+                p["mass::B_s"]           = 5.366;
+                p["decay-constant::B_u"] = 0.1905;
+                p["decay-constant::B_d"] = 0.8940;
+                p["decay-constant::B_s"] = 0.2307;
+                // SM test for the B^-
+                {
+                    static const double eps = 1.0e-6;
+
+                    Options oo
+                    {
+                        { "model", "CKM" },
+                        { "q",     "u"   }
+                    };
+
+                    Lifetime d(p, oo);
+
+                    // compare to known decay width of
+                    // (1.638 +/- 0.004 ps)^-1 = (0.6105 +/- 0.0015) ps^-1
+                    TEST_CHECK_NEARLY_EQUAL(d.decay_width_dbcu(), 0.0153181,   eps);
+                    TEST_CHECK_NEARLY_EQUAL(d.decay_width_sbcu(), 0.000778238, eps);
+                }
+                // SM test for the B_d
+                {
+                    static const double eps = 1.0e-7;
+
+                    Options oo
+                    {
+                        { "model", "CKM" },
+                        { "q",     "d"   }
+                    };
+
+                    Lifetime d(p, oo);
+
+                    // compare to known decay width of
+                    // (1.519 +/- 0.004 ps)^-1
```

Comment:
Can you please add the value and uncertainty in units of inverse picoseconds, just like in line 76?